packages/@aws-cdk/aws-codepipeline/lib/jenkins-actions.ts

```diff
@@ -0,0 +1,120 @@
+import cpapi = require('@aws-cdk/aws-codepipeline-api');
+import cdk = require('@aws-cdk/cdk');
+import { IJenkinsProvider, jenkinsArtifactsBounds } from "./jenkins-provider";
+
+/**
+ * Common construction properties of all Jenkins Pipeline Actions.
+ */
+export interface CommonJenkinsActionProps extends cpapi.CommonActionProps {
+  /**
+   * The name of the project (sometimes also called job, or task)
+   * on your Jenkins installation that will be invoked by this Action.
+   *
+   * @example 'MyJob'
+   */
+  projectName: string;
+
+  /**
+   * The source to use as input for this build.
+   *
+   * @default CodePipeline will use the output of the last Action from a previous Stage as input
+   */
+  inputArtifact?: cpapi.Artifact;
+}
+
+/**
+ * Common properties for creating {@link JenkinsBuildAction} -
+ * either directly, through its constructor,
+ * or through {@link JenkinsBuildProvider#addToPipeline}.
+ */
+export interface CommonJenkinsBuildActionProps extends CommonJenkinsActionProps {
+  /**
+   * The name of the build's output artifact.
+   *
+   * @default an auto-generated name will be used
+   */
+  outputArtifactName?: string;
+}
+
+/**
+ * Construction properties of {@link JenkinsBuildAction}.
+ */
+export interface JenkinsBuildActionProps extends CommonJenkinsBuildActionProps,
+    cpapi.CommonActionConstructProps {
+  /**
+   * The Jenkins Provider for this Action.
+   */
+  jenkinsProvider: IJenkinsProvider;
+}
+
+/**
+ * Jenkins build CodePipeline Action.
+ *
+ * @see https://docs.aws.amazon.com/codepipeline/latest/userguide/tutorials-four-stage-pipeline.html
+ */
+export class JenkinsBuildAction extends cpapi.BuildAction {
+  constructor(parent: cdk.Construct, name: string, props: JenkinsBuildActionProps) {
+    super(parent, name, {
+      provider: props.jenkinsProvider.providerName,
+      owner: 'Custom',
+      artifactBounds: jenkinsArtifactsBounds,
+      version: props.jenkinsProvider.version,
+      configuration: {
+        ProjectName: props.projectName,
+      },
+      ...props,
+    });
+
+    props.jenkinsProvider._includeBuildVariant();
+  }
+}
+
+/**
+ * Common properties for creating {@link JenkinsTestAction} -
+ * either directly, through its constructor,
+ * or through {@link JenkinsTestProvider#addToPipeline}.
+ */
+export interface CommonJenkinsTestActionProps extends CommonJenkinsActionProps {
```

Comment:
I find the prefix `Common` confusing here, since it's not shared, but I see what you're doing.

I've taken to calling these `BasicXxxProps`.

Reply:
Done.